Adds a function which dynamically gets the names of all databases fro…

geek-kb opened the pull request:
…m the source region

pbudzon commented:
Hi @geek-kb 

Thanks for the pull request! The code was actually originally doing exactly that, a previous pull request (#1) introduced a list. Can we maybe add a support for an env variable that will hold a list of databases to apply the code to, and use all databases if the list is empty/not set? I think that would make everyone happy, and a change to the list of databases won't require a new code deployment.

Ideally this code can also be triggered by SNS after "backup" event has been fired from RDS and so the snapshot automatically created by RDS each day can be almost immediately copied to another region. We could technically support all 3 options, I'll add the SNS "detection" later on (or feel free to do so, if you want!).

Thanks.

geek-kb commented:
Hi @pbudzon
You're welcome, I used your code in a solution I created and at the beginning I knew ahead the names of the databases, but later some databases were deleted and the function stopped working so I fixed it on my side and updated your code as well.
But to be honest I'm a beginner in Python and this was my very first code.
I'll do what I can to implement what you've suggested but I can't promise anything at the moment.

Thanks.

pbudzon commented:
I pushed changes to the code (commit 14fd44c7338f3cbe91a7003f6c1141852b7d82ed) that allow for the databases to be optionally specified in CloudFormation parameters. The Lambda will now trigger from RDS backup event for listed databases (or for all, if list is left empty).
That should accommodate all possible use cases :)

Closing PR.
